intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py

```diff
@@ -64,93 +64,102 @@ def __init__(self, persist_dir="./output", process=True, input_path=None,
                                 + '/assets/docs/'
             elif os.path.exists(os.path.join(asset_path, 'docs/')):
                 self.input_path = os.path.join(asset_path, 'docs/')
-            logging.info("The given file path is unavailable, please check and try again!")
         elif isinstance(input_path, List):
             self.input_path = input_path
         else:
             print("The given file path is unavailable, please check and try again!")
-
         assert self.input_path != None, "Should gave an input path!"
+
+        try:
+            if "instruct" in embedding_model:
+                self.embeddings = HuggingFaceInstructEmbeddings(model_name=embedding_model)
+            elif "bge" in embedding_model:
+                self.embeddings = HuggingFaceBgeEmbeddings(
+                    model_name=embedding_model,
+                    encode_kwargs={'normalize_embeddings': True},
+                    query_instruction="Represent this sentence for searching relevant passages:")
+            elif "Google" == embedding_model:
+                self.embeddings = GooglePalmEmbeddings()
+            else:
+                self.embeddings = HuggingFaceEmbeddings(
+                    model_name=embedding_model,
+                    encode_kwargs={"normalize_embeddings": True},
+                )
+        except Exception as e:
+            print("Please selet a proper embedding model")
```

Comment:
shall we use logging? @ftian1 @lvliang-intel

Reply:
yes, here should use logging like this

```python
try:
    ...
except Exception as e:
    logging.exception("message")
```